packages/core/lib/process-request.ts

```diff
@@ -61,6 +68,61 @@ const getExecutableOperation = (document: DocumentNode, operationName?: string):
   return operation;
 };
 
+type AcceptedProtocols = "application/graphql+json" | /* LEGACY */ "application/json" | "text/event-stream" | "multipart/mixed";
+
+type RankedProtocols = Record<AcceptedProtocols, number>;
+
+/**
+ * Returns a map of ranked protocols. Use it for determining which protocol should be used
+ * @param accept Accept header string
+ * @returns
+ */
+const getRankedProtocols = (accept: unknown, contentType: unknown) => {
```

Comment:
can we please have a few unit tests for this function? it does some parsing and transformations to the input, so I'm kinda concerned when we don't have tests for it.

Reply:
6d784cfcb12a0667625bb242b1b903a559b35e81